packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -486,14 +493,14 @@ export abstract class BucketBase extends Resource implements IBucket {
    */
   protected abstract disallowPublicAccess?: boolean;
 
-  private readonly notifications: BucketNotifications;
+  private notifications?: BucketNotifications;
+
+  protected notificationsHandlerRole?: iam.IRole;
+
+  protected handlerRole?: iam.IRole;
```

Comment:
Is this used anywhere? Why do we need it?

Reply:
Oh, this is leftover code from previous refactorings. Thanks for catchings this.